src/compiler/checker.ts

```diff
@@ -20708,9 +20708,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                                     const propDeclaration = prop.valueDeclaration as ObjectLiteralElementLike;
                                     Debug.assertNode(propDeclaration, isObjectLiteralElementLike);
 
-                                    errorNode = propDeclaration;
-
                                     const name = propDeclaration.name!;
+                                    errorNode = name;

```

Comment:
this basically uses the same strategy to overwrite the `errorNode` as the one used by the JSX-related code path here:
https://github.dev/microsoft/TypeScript/blob/6dbec02a88156121c24a152d1e8e5e82d52fdc1d/src/compiler/checker.ts#L20691